packages/block-library/src/comment-template/index.php

```diff
@@ -25,13 +25,14 @@ function block_core_comment_template_render_comments( $comments, $block ) {
 
 	$content = '';
 	foreach ( $comments as $comment ) {
-
-		$block_content = ( new WP_Block(
-			$block->parsed_block,
```

Comment:
One thing I've noticed while working on #50103 is that if a `render_block_data` filter modifies `$parsed_block`, the old code here would make sure that the block instance itself is updated accordingly. E.g. if I have a

```php
function gutenberg_auto_insert_child_block( $parsed_block, $source_block, $parent_block ) {
	if ( 'core/comment-template' === $parsed_block['blockName'] ) {
		$inserted_block_markup = <<<END
<!-- wp:social-links -->
<ul class="wp-block-social-links"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /--></ul>
<!-- /wp:social-links -->'
END;
		$inserted_blocks = parse_blocks( $inserted_block_markup );

		$parsed_block['innerBlocks'][] = $inserted_blocks[0];
);
	}
	return $parsed_block;
}
add_filter( 'render_block_data', 'gutenberg_auto_insert_child_block', 10, 3 );
```

then a `render_block` filter like

```php
function gutenberg_auto_insert_blocks( $block_content, $block, $instance ) {
	if ( 'core/comment-template' === $block['blockName'] ) {
		echo count( $instance->inner_blocks ) . "\n";
		echo count( $block['innerBlocks'] ) . "\n";
	}

	return $block_content;
}
add_filter( 'render_block', 'gutenberg_auto_insert_blocks', 10, 3 );
```

would return

```
2
2
```

With this PR applied however, `$instance->inner_blocks` is not updated to include the newly inserted inner block from the parsed block:

```
1
2
```

Reply:
In practice, this could lead to issues with filters such as [this](https://github.com/WordPress/gutenberg/blob/82b98438f62f4c8181aabc9123b6f68ce25c52cd/packages/block-library/src/gallery/index.php#L8-L33).

So if we need to restore the re-parsing, we could maybe try something like

```diff
diff --git a/packages/block-library/src/comment-template/index.php b/packages/block-library/src/comment-template/index.php
index 6a8698f60f..d3ffc67cf8 100644
--- a/packages/block-library/src/comment-template/index.php
+++ b/packages/block-library/src/comment-template/index.php
@@ -31,7 +31,7 @@ function block_core_comment_template_render_comments( $comments, $block ) {
                        return $context;
                };
                add_filter( 'render_block_context', $filter_block_context );
-               $block_content = $block->render( array( 'dynamic' => false ) );
+               $block_content = ( new WP_Block( $block->parsed_block ) )->render( array( 'dynamic' => false ) );
                remove_filter( 'render_block_context', $filter_block_context );
 
                $children = $comment->get_children();
```

(untested)